### DISCUSSION TITLE
Where does LLMWare fit into the ecosystem?

### DISCUSSION THREAD

**Discussion by hughesadam87** [llmware-ai/llmware#74] [General]
Hi,

I found `llmware` through this reddit thread (https://www.reddit.com/r/vectordatabase/comments/17lhqri/comment/k7nbgui/?context=3), where the goal is to understand the landscape of open-source RAG.  

From what I've gotten from your docs, llmware is more purpose-built than langchain and offers higher-level capabilities.  So for example, if I wanted to build a RAG solution for my company, but didn't want to buy some managed service, I'd get there faster with llmware than started from ground-level Langchain?  Is this correct?

Is there a vision or roadmap for all that will be contained in llmware?  For example, adding access controls into a RAG system so that users can't see context from documents that would otherwise be private (think a medical app with HIPPA data).  Is this in scope for llmware?

Is llmware backed by a company or just a hobby project?

Thanks

**Comment by NYDocutest** [comment]
Thank you so much for your question! LLMWare is the open source version of Ai Bloks' LLM-based solution that includes the RAG pipeline you found mentioned in Reddit as well as open source contributions in Hugging Face including domain specific embedding models, instruct test datasets and small LLMs that have been fine-tuned for RAG.

We developed our LLMWare RAG pipeline to be very easy to use for beginners (one Redditor said our RAG was "dead simple" to get started) but is also an enterprise-grade solution with incredible data ingestion capabilities (we developed our own PDF and Office doc parsers to optimize performance) as well as a lot of other features built in such as fact checking, source citation, guard rails against hallucination, auditability, and many more!

We are generally focused on Legal and Financial Services enterprise use cases for the time being but we have not ruled out medical app use cases for our future pipeline. We are always looking for feedback so if this is a need that is not being addressed then we would be happy to investigate for the future.

As you can hopefully see from the care and pride we take in our open source offerings, it is not a hobby project. With the continued support of the AI community, we hope to grow and keep going!

Please feel free to reach out with any more questions!

**Reply by hughesadam87** [reply]
That's awesome, thank you for the response.
